windows-agent/internal/config/config.go

```diff
@@ -130,6 +138,10 @@ func (c *Config) set(ctx context.Context, field *string, value string) error {
 	old := *field
 	*field = value
 
+	for _, f := range c.notifiers {
+		go f()
```

Comment:
I assume here the callbacks will contain a reference to the Config, so the observers won't need to have the latest Config data passed in the call. Did you consider having such data as a parameter, thus avoiding the need for storing a reference to the Config? That's a common practice in callback systems of that nature, as well as passing copies of the data to prevent races is also common.

Finally, it's quite common as well to pass the old and the new data so the observer compares the bits it's interested in before taking any action , but I think that would be too much to our need, right.

Reply:
I did consider it, but figured it was a bit silly:
```go
config.Notify(func(conf *Config) {
    // use conf
})
```
You already had access to the config to make the Notify call, so it seems a bit redundant. Config is always a pointer so it's always up to date.

I do not pass what the diff is because that'd introduce too much logic (at least for now, we only have two pieces of data anyways).